addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2536,21 +2931,29 @@ registry.background = SnippetOptionWidget.extend({
      * Handles a background change.
      *
      * @see this.selectClass for parameters
+     * @param isReset whether the method is being called to reset a preview.
      */
-    background: async function (previewMode, widgetValue, params) {
-        if (previewMode === 'reset') {
-            return this._setCustomBackground(this.__customImageSrc, previewMode);
-        }
-        if (!previewMode) {
+    background: async function (previewMode, widgetValue, params, isReset) {
```

Comment:
What's that parameter? Just use `previewMode === 'reset'`?

Reply:
No no, see, in `previewMode = reset`, we call  `_setCustomBackground`, which calls `background` back with `previewMode = false` to actually reset the background, but it's actually a reset despite previewMode suggesting otherwise. It's ugly but there was a quite a few edge cases and interactions that made it non-trivial to rewrite so that it doesn't do it that way. I can spend some time on this to clean it up instead if you'd like but when I first tried I ran into some issues and decided to use this workaround for now.

Reply:
Yes please retry. Maybe it's just about extracting some content of the `background` method into a new function free of the "previewMode" notion so that ` _setCustomBackground` does actually calls that new function instead?

Reply:
I managed to inline all the logic, apparently _setCustomBackground was only called from there, probably was a relic from a previous version of the option system.